### ISSUE TITLE
Definition of Epoch with Mask RCNN.

### ISSUE THREAD

**Issue by ecm200** [facebookresearch/detectron2#1349]
I am relatively new to Mask RCNN architecture so just finding my feet.

What is the definition of an Epoch with training Mask RCNN?
Is it even a sensible term in Mask RCNN training?

In image classification, an Epoch is defined as a set of mini batches over the training set of images generally **without replacement**, such that per Epoch each image is sampled once. Validation is usually performed after a single complete pass over the training set. That is, at the end of each Epoch.

I understand the following:

- Different branches of the network have different mini-batch sizes.

- Network backbone mini-batch size is controlled by SOLVER.IMS_PER_BATCH. Default is 2 per GPU. 

- Following literature, learning rate should be linearly scaled from base rate of 16 images per minibatch and a LR of 0.02.

- Default Training Sampler shuffles training set and sample **WITH REPLACEMENT**.

Given all this, is there a definition of Epoch?

**With replacement** sampling implies an infinite stream of images, with no guarantee that all images will be sampled after iterations of number [num_images /  mini_batch_size].

If this is the case, then what is the common thinking about the logic of defining **iteration periods for running validation and checkpointing**.

**Comment by ppwwyyxx** [comment]
There is no definition of epoch in detectron2. You can define it however you want if you need.

In certain configs, epoch is defined as a pass over all the training images.

**Comment by deepakksingh** [comment]
Hello @ecm200 , did you resolve or find any workaround for this issue. I'm having similar requirement too.
